[19.0][MIG] template_content_swapper: Migration to 19.0

AungKoKoLin1997 opened the pull request:
@qrtl QT5667

AungKoKoLin1997 committed:
[ADD] template_content_swapper

Commit:
[UPD] Update template_content_swapper.pot

OCA-git-bot committed:
[BOT] post-merge updates

AungKoKoLin1997 committed:
[FIX] template_content_swapper
This commit resolves the issue by adding 'report_id', which enables the computation of 'template_id'.
Previously, the problem occurred because 'template_id' is a mandatory field that needs to be calculated before inserting the record.

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (23 of 23 strings)

Translation: server-ux-16.0/server-ux-16.0-template_content_swapper
Translate-URL: https://translation.odoo-community.org/projects/server-ux-16-0/server-ux-16-0-template_content_swapper/it/

AungKoKoLin1997 committed:
[FIX] template_content_swapper: use _render()

This commit moves the content mapping logic from _render_template of ir.ui.view to _render of ir.qweb, which is used more consistently,
ensuring content replacement works in all rendering scenarios.

Commit:
[UPD] Update template_content_swapper.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-ux-16.0/server-ux-16.0-template_content_swapper
Translate-URL: https://translation.odoo-community.org/projects/server-ux-16-0/server-ux-16-0-template_content_swapper/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (23 of 23 strings)

Translation: server-ux-16.0/server-ux-16.0-template_content_swapper
Translate-URL: https://translation.odoo-community.org/projects/server-ux-16-0/server-ux-16-0-template_content_swapper/it/

AungKoKoLin1997 committed:
[IMP] template_content_swapper: pre-commit auto fixes

AungKoKoLin1997 committed:
[MIG] template_content_swapper: Migration to 18.0

AungKoKoLin1997 committed:
[FIX] template_content_swapper: fix column_invisible is not working

Commit:
[UPD] Update template_content_swapper.pot

OCA-git-bot committed:
[BOT] post-merge updates

AungKoKoLin1997 committed:
[MIG] template_content_swapper: Migration to 19.0

AungKoKoLin1997 committed:
[MIG] template_content_swapper: Migration to 19.0

AungKoKoLin1997 committed:
[IMP] template_content_swapper: add feature to use domain for applying some specific records in report pdfs

Co-authored-by: Yoshi Tashiro <tashiro@quartile.co>

AungKoKoLin1997 committed:
[IMP] template_content_swapper: add feature to use domain for applying some specific records in report pdfs

Co-authored-by: Yoshi Tashiro <tashiro@quartile.co>

smorita7749 reviewed (commented):
- Is it acceptable to put HTML to "Content to"?


- I tested the PDF Quote with the following mappings:

| Content From | Content To     |
|---|---|
| Order        | Purchase Order |
| Order Date   | Request Date   |

Then, the result was as follows.

<img width="514" height="179" alt="Image" src="https://github.com/user-attachments/assets/5d77d44f-d5fe-4095-821d-5c334d8bac61" />


I think this cascading behavior could confuse users. At minimum, it would be good to mention it in the README as a known limitation.

smorita7749 reviewed (approved):
Functional review: LGTM

SanamiK reviewed (approved):
Functional review. LGTM.

yostashiro commented on template_content_swapper/__manifest__.py:
```diff
@@ -0,0 +1,17 @@
+# Copyright 2024 Quartile (https://www.quartile.co)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+{
+    "name": "Template Content Swapper",
```
Please add a summary.

AungKoKoLin1997 replied:
Done!

nobuQuartile reviewed (approved):
Code review and functional review: LGTM

BhaveshHeliconia reviewed (approved):
LGTM!

yostashiro commented:
@AungKoKoLin1997 I assume we should technically split the last commit into two: api.model and tests

AungKoKoLin1997 committed:
[FIX] template_content_swapper: restore @api.model on _render override

The domain feature commit dropped the @api.model decorator (and the api
import) from the _render override, diverging from the core
ir.qweb._render signature. Restore it.

AungKoKoLin1997 committed:
[IMP] template_content_swapper: test multi-record domain render

The domain feature's multi-record path (_apply_article_mappings, the
len(articles) > 1 branch) had no test: the existing domain test only
renders a single-record preview, so the per-article matching and the
match_map membership filtering across several data-oe-id blocks were
never exercised. Add a test that applies a domain mapping to a two-article
document and asserts only the matching record is swapped.

AungKoKoLin1997 commented:
> I assume we should technically split the last commit into two: api.model and tests

@yostashiro Done!

yostashiro reviewed (approved):
LGTM. 👍
